plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -840,51 +839,17 @@ public void onFailure(Throwable t) {
     }
 
     /**
-     * Used by {@link #loadProgramAsync(File)} to propagate a failure to load the persisted execution state.
-     * <p>
-     * Let the workflow interrupt by throwing an exception that indicates how it failed.
+     * Used to propagate a failure to load the persisted execution state.
      * @param promise same as {@link #programPromise} but more strongly typed
      */
     private void loadProgramFailed(final Throwable problem, SettableFuture<CpsThreadGroup> promise) {
-        FlowHead head;
-
-        synchronized(this) {
-            if (heads == null || heads.isEmpty()) {
-                head = null;
-            } else {
-                head = getFirstHead();
-            }
-        }
-
-        if (head==null) {
-            // something went catastrophically wrong and there's no live head. fake one
-            head = new FlowHead(this);
-            try {
-                head.newStartNode(new FlowStartNode(this, iotaStr()));
-            } catch (IOException e) {
-                LOGGER.log(Level.FINE, "Failed to persist", e);
-            }
+        try {
+            Functions.printStackTrace(problem, owner.getListener().getLogger());
+        } catch (Exception x) {
+            LOGGER.log(Level.WARNING, "failed to log problem to " + owner, x);
```

Comment:
This overload exists fwiw:

```suggestion
            LOGGER.log(Level.WARNING, x, () -> "failed to log problem to " + owner);
```